include/awkward/Index.h

```diff
@@ -220,7 +222,7 @@ namespace awkward {
     /// @brief See #ptr_lib
     const kernel::lib ptr_lib_;
     /// @brief See #offset.
-    const int64_t offset_;
+    const int64_t offset_ = 0;
```

Comment:
Is this supposed to be related to the bug? If there's any way for `offset_` to be uninitialized in an `IndexOf<T>` instance, then that itself is a bug that needs to be addressed at its source (assuming `0` is a bad idea: it must always be given when the instance is constructed).